src/cloudant/result.py

```diff
@@ -347,24 +347,27 @@ def __iter__(self):
         except ValueError:
             raise ResultException(104, self._page_size)
 
-        skip = 0
+        last = self.options.pop('startkey', '')
         while True:
             response = self._ref(
-                limit=self._page_size,
-                skip=skip,
+                limit=self._page_size + 1,
+                startkey=last,
                 **self.options
             )
             result = self._parse_data(response)
-            skip += self._page_size
             if result:
+                last = result.pop()
                 for row in result:
                     yield row
                 if len(result) < self._page_size:
```

Comment:
Change to `if len(result) < self._page_size + 1`?

Also, I think a comment here would be good. Something like:

> We expect `len(result) = self._page_size + 1` results, if we have ` self._page_size` or less it means we are on the last page and need to return the last result.

Reply:
you're right, this way we save a hit on the worst case when ndocs%pagesize==0 ... neat !